tiledb/cloud/group.py

```diff
@@ -0,0 +1,127 @@
+from typing import Dict, List
+
+from tiledb.cloud import client
+from tiledb.cloud import tiledb_cloud_error
+from tiledb.cloud import utils
+from tiledb.cloud.array import delete_array
+from tiledb.cloud.rest_api import ApiException as GenApiException
+
+# Re-export for compatibility.
+split_uri = utils.split_uri
+
+
+def get_group(uri):
+    """
+    Returns the cloud metadata
+
+    :return: metadata object
+    """
+    (namespace, group_name) = split_uri(uri)
+    api_instance = client.client.groups_api
+
+    try:
+        return api_instance.get_group(namespace, group_name, async_req=async_req)
+    except GenApiException as exc:
+        raise tiledb_cloud_error.check_exc(exc) from None
+
+
+def delete_group_recursively(uri, dry_run=False, verbose=False):
+    """
+    Deletes the group object, after recursively deregistering/deleting child array elemens
+    and deleting child group elements.
+
+    :param uri: TileDB Cloud URI of the group to be deleted
+
+    :param dry_run: If true, print what would be deleted without doing the deletion.
+
+    :param verbose: If true, print what is being deleted. The `dry_run` flag
+    implies `verbose`.
+
+    :return success or error
+    """
+
+    if dry_run:
+        verbose = True
+
+    (namespace, group_name) = split_uri(uri)
+
+    # Get all members before deleting any, since we don't want to be deleting
+    # things we're iterating over.
+    group_element_tiledb_uris = _get_group_element_tiledb_uris(uri)
+
+    for e in group_element_tiledb_uris["arrays"]:
+        if verbose:
+            print("DELETE ARRAY", e)
+        if not dry_run:
+            delete_array(e)
+    for e in group_element_tiledb_uris["groups"]:
+        # Depth-first traversal
+        delete_group_recursively(e, dry_run=dry_run, verbose=verbose)
+        if verbose:
+            print("DELETE GROUP", e)
+        if not dry_run:
+            delete_group(e)
+
+
+def delete_group(uri: str) -> None:
+    """
+    Deletes the group. The assets are not deleted nor are not relocated to any other group.
+    This method makes a synchronous HTTP request by default.
+
+    You probably want `delete_group_recursively`.
+
+    :param uri: TileDB Cloud URI of the group to be deleted
+    """
+    (namespace, group_name) = split_uri(uri)
+    api_instance = client.client.groups_api
+    try:
+        return api_instance.delete_group(namespace, group_name)
+    except GenApiException as exc:
+        raise tiledb_cloud_error.check_exc(exc) from None
+
+
+def _get_group_element_tiledb_uris(uri: str) -> Dict[str, List[str]]:
```

Comment:
Instead of a dict, I would return either a pair of lists here. While in most cases I am wary of using undifferentiated pairs, in this case since it's an internal function, I think it's probably fine given appropriate naming, something like:

`def _subgroups_and_arrays(uri: str) -> Tuple[List[str], List[str]]:`